src/Builder/BuilderEncoder.php

```diff
@@ -182,7 +183,19 @@ private function encodeAsDollarObject(OperatorInterface $value): stdClass
                 continue;
             }
 
-            $result->{'$' . $key} = $this->recursiveEncode($val);
+            $val = $this->recursiveEncode($val);
+
+            if ($key === 'geometry') {
```

Comment:
We may want to consider introducing a special encoder for geo operators to extract this special logic.

Reply:
For everything related to geometry, we need to be compatible with a GeoJSON library instead of creating a `$geometry` object.

Reply:
I agree. Apart from a couple of libraries that saw their last updates 9 years ago, the only PHP libraries are found are [jmikola/geojson](https://github.com/jmikola/geojson) which is supported in Doctrine ODM, and [brick/geo](https://github.com/brick/geo) which is still in a pre-release (0.x) state but saw its last update a little more than a year ago. I'll defer to @jmikola's expertise on this.

Reply:
Tracking in [PHPLIB-1364](https://jira.mongodb.org/browse/PHPLIB-1364)